ecosystem/typescript/sdk_v2/src/crypto/secp256k1.ts

```diff
@@ -0,0 +1,239 @@
+   * @returns true if the signature is valid
+   */
+  verifySignature(args: { message: HexInput; signature: Secp256k1Signature }): boolean {
+    const { message, signature } = args;
+    const msgHex = Hex.fromHexInput({ hexInput: message }).toUint8Array();
+    const sha256Message = sha256(msgHex);
+    const rawSignature = signature.toUint8Array();
+    return secp256k1.verify(rawSignature, sha256Message, this.toUint8Array());
+  }
+
+  serialize(serializer: Serializer): void {
+    serializer.serializeBytes(this.key.toUint8Array());
+  }
+
+  static deserialize(deserializer: Deserializer): PublicKey {
+    const bytes = deserializer.deserializeBytes();
+    return new Secp256k1PublicKey({ hexInput: bytes });
+  }
+
+  // eslint-disable-next-line class-methods-use-this,@typescript-eslint/no-unused-vars
+  deserialize(deserializer: Deserializer): PublicKey {
+    throw new Error("Not implemented");
+  }
+}
+
+/**
+ * A Secp256k1 ecdsa private key
+ */
+export class Secp256k1PrivateKey extends PrivateKey {
+  /**
+   * Length of Secp256k1 ecdsa private key
+   */
+  static readonly LENGTH: number = 32;
+
+  /**
+   * The private key bytes
+   * @private
+   */
+  private readonly key: Hex;
+
+  /**
+   * Create a new PrivateKey instance from a Uint8Array or String.
+   *
+   * @param args.hexInput A HexInput (string or Uint8Array)
+   */
+  constructor(args: { hexInput: HexInput }) {
+    super();
+
+    const privateKeyHex = Hex.fromHexInput(args);
+    if (privateKeyHex.toUint8Array().length !== Secp256k1PrivateKey.LENGTH) {
+      throw new Error(`PrivateKey length should be ${Secp256k1PrivateKey.LENGTH}`);
+    }
+
+    this.key = privateKeyHex;
+  }
+
+  /**
+   * Get the private key in bytes (Uint8Array).
+   *
+   * @returns
+   */
+  toUint8Array(): Uint8Array {
+    return this.key.toUint8Array();
+  }
+
+  /**
+   * Get the private key as a hex string with the 0x prefix.
+   *
+   * @returns string representation of the private key
+   */
+  toString(): string {
+    return this.key.toString();
+  }
+
+  /**
+   * Sign the given message with the private key.
+   * The `lowS` option is default to `true`
+   * lowS: true prohibits signatures which have (sig.s >= CURVE.n/2n) and is compatible with BTC/ETH.
+   *
+   * @param args.message in HexInput format
+   * @returns Signature
+   */
+  sign(args: { message: HexInput }): Secp256k1Signature {
+    const msgHex = Hex.fromHexInput({ hexInput: args.message });
+    const sha256Message = sha256(msgHex.toUint8Array());
+    const signature = secp256k1.sign(sha256Message, this.key.toUint8Array());
+    return new Secp256k1Signature({ hexInput: signature.toCompactRawBytes() });
+  }
+
+  serialize(serializer: Serializer): void {
+    serializer.serializeBytes(this.toUint8Array());
+  }
+
+  // TODO: Update this in interface to be static, then remove this method
+  deserialize(deserializer: Deserializer): Secp256k1PrivateKey {
+    throw new Error("Method not implemented.");
+  }
```

Comment:
```suggestion
```

You should be able to remove this now, I just merged the change!

Reply:
Ahh sounds good captain!

Reply:
@xbtmatt Hmm wait, looks like its not merged yet? - https://github.com/aptos-labs/aptos-core/pull/10307

Reply:
@0xjinn Yeah I was fighting the cicd pipeline all weekend haha. it looks like it's finally working, but I wanted to update something before merging- trying to merge it in in the next couple hours